cmd/minishift/cmd/start.go

```diff
@@ -324,6 +327,22 @@ func startHost(libMachineClient *libmachine.Client) *host.Host {
 		fmt.Println("   Disk size:", units.HumanSize(float64(machineConfig.DiskSize*units.MB)))
 	}
 
+	// Experimental features
+	if hasEnabledExperimental {
+		networkSettings := minishiftNetwork.NetworkSettings{
+			Device:    viper.GetString(configCmd.NetworkDevice.Name),
+			IPAddress: viper.GetString(configCmd.IPAddress.Name),
+			Netmask:   viper.GetString(configCmd.Netmask.Name),
+			Gateway:   viper.GetString(configCmd.Gateway.Name),
+			DNS1:      viper.GetString(configCmd.Nameserver.Name),
+		}
+
+		// Configure networking on startup only works on Hyper-V
+		if machineConfig.VMDriver == "hyperv" && networkSettings.IPAddress != "" {
+			minishiftNetwork.ConfigureNetworking(constants.MachineName, networkSettings)
```

Comment:
Although driver is checked, only the `_windows.go` file will have an implementation.